File: exercises/small_problems/easy7/end_is_near.rb
# START
# GIVEN a string at least two words long
# RETURN the second to last word
# END

# START
# GIVEN a string of any number of words
# RETURN the middle word
# END

# ---EDGE CASES
#   even number of words : return the middle two words
#   empty string: return empty string
#   one word string: return that word


def penultimate(str)
  str.split[-2]
end

def middle_word(str)
  return str unless str.split.size >= 2

  new_str = ""
  arr = str.split
  middle = arr.size / 2
  if arr.size.even?
    new_str << arr[middle - 1] + " " + arr[middle]
  else
    new_str = arr[middle]
  end

  new_str
end

p penultimate('last word') == 'last'
p penultimate('Launch School is great!') == 'is'

p middle_word('what is the last word, yo?') == 'the last'
p middle_word('Launch School is really great!') == 'is'
p middle_word('The Dude Abides') == 'Dude'
p middle_word('Launch') == 'Launch'
p middle_word('') == ''

File: exercises/210_JS_Fun/objects/endIsNear.js
// this function is trying to access a non-existant index

function penultimate(string) {
  return string.split(' ')[-2];
}

//this function corrects for that

function penultimate2(string) {
  const stringArr = string.split(' ');
  return stringArr[stringArr.length - 2];
}

console.log(penultimate('last word')); // expected: "last"
console.log(penultimate('Launch School is great!')); // expected: "is"

console.log(penultimate2('last word')); // expected: "last"
console.log(penultimate2('Launch School is great!')); // expected: "is"
